### PR TITLE
feat(tabset): add nav justification and orientation

### PR DESCRIPTION
**Pull request by ktriek**
- Nav Justification: 
  - `start`: align left `.justify-content-start`
  - `center`: align center `.justify-content-center`
  - `end`: align right `.justify-content-end`
  - `fill`:  All horizontal space will be occupied by nav links `.nav-fill`
  - `justified`: All horizontal space will be occupied by nav links and every nav item will be the same width   `.nav-justified`
- Nav orientation:
  - `horizontal`
  - `vertical`

---
Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

File: src/tabset/tabset.ts [hunk 1]
```diff
@@ -105,9 +107,24 @@ export class NgbTabset implements AfterContentChecked {
...
+   * 'justified'
+   * The default value is 'start'.
    */
-  @Input() justify: 'start' | 'center' | 'end';
+  @Input()
+  set justify(className: 'start' | 'center' | 'end' | 'fill' | 'justified') {
+    if (className === 'fill' || className === 'justified') {
+      this.justifyClass = 'nav-' + className;
+    } else {
+      this.justifyClass = 'justify-content-' + className;
```
Comment (wesleycho) [inline src/tabset/tabset.ts]: Could pretty the class setting to
```ts
this.justifyClass = `nav-${className}`;
```
and
```ts
this.justifyClass = `justify-content-${className}`;
```

**Comment by wesleycho** [comment]
Looks like this needs to be rebased, but otherwise should be good.

**Comment by JordanGS** [comment]
someone wanna rebase it and added it if it's good (assuming that's the only thing stopping the commit)? I'd do it myself if i could but @ktriek hasn't made a comment in almost 2 months.

**Comment by korn3l** [comment]
This seems to be a very useful feature. Thanks

**Comment by lsturtevant** [comment]
What's the status of this PR?  I need this to be able to use ng-bootstrap for tabs since it finally implements missing features.

**Comment by trafalgarDWaterLaw** [comment]
Hi, when is this coming along?

**Comment by pkozlowski-opensource** [comment]
Merged, thnx a lot @ktriek ! 

Sorry it took as so long but we were busy with other items and waiting for Bootstrap 4 beta to see if there are no more changes coming. 

Once again, thnx for the PR - we love PRs with tests, demo updates and clean code. Keep them coming!

**Comment by ktriek** [comment]
Thanks @pkozlowski-opensource for your time, I really appreciate what you are doing.

**Comment by sbatezat** [comment]
Tabset is deprecated in favor of nav and it seems that nav-justified is not working on nav.

`<ul ngbNav [justify]="justified" class="nav-pills">
`
or 

`<ul ngbNav class="nav-pills nav-justified">`

does not produce the expected behavior.

**Pull request closed without merging**